nltk.internals.compile_regexp_to_noncapturing doctest is broken

In Python 2.7:

```
======================================================================
FAIL: Doctest: nltk.internals.compile_regexp_to_noncapturing
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.5/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 2201, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for nltk.internals.compile_regexp_to_noncapturing
  File "/Users/kmike/svn/nltk/nltk/__init__.py", line 38, in compile_regexp_to_noncapturing

----------------------------------------------------------------------
File "/Users/kmike/svn/nltk/nltk/__init__.py", line 44, in nltk.internals.compile_regexp_to_noncapturing
Failed example:
    compile_regexp_to_noncapturing('ab(c(x+)(z*))?d')
Expected:
    'ab(?:c(?:x+)(?:z*))?d'
Got:
    <_sre.SRE_Pattern object at 0x7fa09e70b2e0>
```

--- a/nltk/internals.py
+++ b/nltk/internals.py
@@ -31,12 +31,8 @@ from nltk import compat
 
 def compile_regexp_to_noncapturing(pattern, flags=0):
     """
-    Convert all grouping parentheses in the given regexp pattern to
-    non-capturing groups, and return the result.  E.g.:
-
-        >>> from nltk.internals import compile_regexp_to_noncapturing
-        >>> compile_regexp_to_noncapturing('ab(c(x+)(z*))?d')
-        'ab(?:c(?:x+)(?:z*))?d'
+    Compile the regexp pattern after switching all grouping parentheses
+    in the given regexp pattern to non-capturing groups.
 
     :type pattern: str
     :rtype: str

--- a/nltk/internals.py
+++ b/nltk/internals.py
@@ -46,7 +46,7 @@ def compile_regexp_to_noncapturing(pattern, flags=0):
         for key, value in parsed_pattern.data:
             if key == sre_constants.SUBPATTERN:
                 index, subpattern = value
-                value = (None, convert_regexp_to_noncapturing(subpattern))
+                value = (None, convert_regexp_to_noncapturing_parsed(subpattern))
             elif key == sre_constants.GROUPREF:
                 raise ValueError('Regular expressions with back-references are not supported: {0}'.format(pattern))
             res_data.append((key, value))

--- a/nltk/internals.py
+++ b/nltk/internals.py
@@ -51,7 +51,7 @@ def compile_regexp_to_noncapturing(pattern, flags=0):
         parsed_pattern.pattern.groupdict = {}
         return parsed_pattern
 
-    return sre_compile.compile(convert_regexp_to_noncapturing_parsed(sre_parse.parse(pattern)))
+    return sre_compile.compile(convert_regexp_to_noncapturing_parsed(sre_parse.parse(pattern)), flags=flags)
 
 
 ##########################################################################

